extensions/indexes/lucene/src/main/java/org/exist/indexing/lucene/LuceneFacetConfig.java

```diff
@@ -92,6 +98,19 @@ protected void processResult(Sequence result, Document luceneDoc) throws XPathEx
         }
     }
 
+    private void createHierarchicalFacet(Document luceneDoc, Sequence seq) throws XPathException {
+        final List<String> paths = new ArrayList<>(seq.getItemCount());
+        for (SequenceIterator i = seq.unorderedIterator(); i.hasNext(); ) {
+            final String value = i.nextItem().getStringValue();
+            if (value.length() > 0) {
```

Comment:
functional identical to !String#isEmpty() ; the current is faster for sure.
Is there a NPE risk?

Reply:
In line with XQuery specs, `getStringValue()` on an item should not return null.

Reply:
I think @dizzzz is right here.